tiledb/common/thread_pool/thread_pool.cc

```diff
@@ -67,25 +57,25 @@ Status ThreadPool::init(size_t n) {
       } catch (const std::system_error& e) {
         if (e.code() != std::errc::resource_unavailable_try_again ||
             tries == 0) {
-          st = Status_ThreadPoolError(
+          auto st = Status_ThreadPoolError(
               "Error initializing thread pool of concurrency level " +
               std::to_string(concurrency_level_) + "; " + e.what());
           LOG_STATUS(st);
-          break;
+          shutdown();
+          throw;
```

Comment:
Should we throw anything more meaningful here and below?

Reply:
I think that would be a good idea.  I have another story open (15825) about dealing better with thread pool errors.